packages/itwinui-css/src/utils/icon/icon-component.scss

```diff
@@ -0,0 +1,43 @@
+// Copyright (c) Bentley Systems, Incorporated. All rights reserved.
+// See LICENSE.md in the project root for license terms and full copyright notice.
+@use 'sass:math';
+@use 'sass:map';
+
+$icon-sizes: (
+  's': #{math.div(12, 16)}rem,
+  'm': #{math.div(16, 16)}rem,
+  'l': #{math.div(24, 16)}rem,
+  'auto': #{math.div(16, 14)}em,
+);
+
+@mixin iui-svg-icon {
+  --_iui-svg-size: #{map.get($icon-sizes, m)};
+  --_iui-svg-fill: var(--iui-color-icon-muted);
+
+  display: inline-flex;
+  flex-shrink: 0;
+
+  &,
+  svg {
+    width: var(--_iui-svg-size);
+    height: var(--_iui-svg-size);
+    fill: var(--_iui-svg-fill);
```

Comment:
Is there a way to supply a custom `fill` when using `<Icon>` for just the default sizing? I guess we have to manually overwrite the `--_iui-svg-fill` CSS variable, but is there an easier way?

Because when I tried to supply a custom fill to `<SvgAdd fill='red'/>`, it was not used.

```tsx
<Icon>
  <SvgAdd fill='red' />
</Icon>
```

<!-- ![image](https://user-images.githubusercontent.com/45748283/211069767-6dfcf8dc-ef3f-415e-8100-7609fc30fc60.png) -->
![image](https://user-images.githubusercontent.com/45748283/211069992-96ec7ac8-ca2d-4c94-9202-27e699e42c80.png)

Reply:
Just realized, we can do that with inline style. To confirm, is this the correct way?

```tsx
<Icon>
  <SvgAdd style={{ fill: 'red' }} />
</Icon>
```